### DISCUSSION TITLE
Send data every 20 minutes

### DISCUSSION THREAD

**Discussion by sgaeumann** [cyberman54/ESP32-Paxcounter#957] [Q&A] [answered]
Hello.

My goal is to send the pax counters every 20 minutes (more or less). In the `paxcounter.conf` file, there's a possibility to change the `SENDCYCLE` value, however it is limited to 255 (which would send every 2 minutes approximatively). I think the solution would be to modify the `SLEEPCYCLE` value.

`#define SENDCYCLE   60      // payload send cycle [seconds/2], 0 .. 255`
`#define SLEEPCYCLE   0       // sleep time after a send cycle [seconds/10], 0 .. 65535; 0 means no sleep [default = 0]`

* Is the device still sniffing while on `SLEEPCYCLE` ? 
* Or is the sniffing enabled after each cycle and disabled until next cycle?

Thanks in advance.

**Comment by cyberman54** [comment, marked as answer]
_>Or is the sniffing enabled after each cycle and disabled until next cycle?_
This is the way how it works.

SENDCYCLE defines the sniffing period while the device is sniffing, and at timeout sends the result.
SLEEPCYCLE defines the pausing period between sniffing cycles. While pausing the device is set to sleep mode to reduce power consumption. Wifi & BLE are switched off during sleep. Thus, no sniffing.

**Comment by cyberman54** [comment]
Note: SENDCYCLE = 255 means a sniffing period of 255 * 2 = 510 seconds = 8,5 minutes continously sniffing. Not 2 minutes.

**Comment by proffalken** [comment]
Ah, now I understand this (I think!), so to send data every 10 minutes, sniffing for two minutes at a time, it would look like this:

```
#define SENDCYCLE 60
#define SLEEPCYCLE 600
```

Is that right?

**Reply by cyberman54** [reply]
yes.

**Reply by proffalken** [reply]
Perfect, thanks!
